node_modules/col-activities/lib/api.js

```diff
@@ -178,26 +178,29 @@ var exportActivities = module.exports.exportActivities = function(ctx, callback)
  * @param  {String}           objectType                          The type of the object on which the activity is taking place. One of `CollabosphereConstants.ACTIVITY.OBJECT_TYPES`
  * @param  {Object}           [metadata]                          Additional metadata that is associated with the activity. For example, when creating a submission activity it might be useful to store the submission id, the attempt number and the attachment ids
  * @param  {User}             [actor]                             The user performing the activity when different than the user earning activity points
+ * @param  {Number}           [assetId]                           The associated asset id when different from the object id (e.g., for comments)
  * @param  {Function}         callback                            Standard callback function
  * @param  {Object}           callback.err                        An error that occurred, if any
  * @param  {Activity}         callback.activity                   The created activity
  */
-var createActivity = module.exports.createActivity = function(course, user, type, objectId, objectType, metadata, actor, callback) {
+var createActivity = module.exports.createActivity = function(course, user, type, objectId, objectType, metadata, actor, assetId, callback) {
```

Comment:
I think it's technically possible in Postgres to join by a value inside of a stored JSON document, but I'm not entirely sure how performant this is. The alternative could be to just generalize `assetId` a bit more (e.g., `targetId`) so it can apply to whiteboards as well?